clojure-mode.el

```diff
@@ -380,7 +380,7 @@ Called by `imenu--generic-function'."
        (2 font-lock-type-face nil t))
       ;; Function definition (anything that starts with def and is not
       ;; listed above)
-      (,(concat "(\\(?:[a-z\.-]+/\\)?\\(def\[a-z\-\]*-?\\)"
+      (,(concat "(\\(?:[a-z\.-]+/\\)?\\(def[^ \r\n\t]*\\)"
```

Comment:
Not really... I'm using `^` for exclusion.
So I'm saying take anything that starts with a `def` and it is followed by anything which is **NOT** a blank 0 or more times. So to be fair should be something like `[^[:space:]]`.
Since different regex engines use different character classes (`[:space:]`, `[:blank:]`, `[:s:]`, `\s`) I wasn't sure about the correct one for elisp and I took reference from line 387 and 389 (just below)